src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -514,18 +523,11 @@ private boolean hasFallThroughComment(DetailAST currentCase, DetailAST nextCase)
      * possible match is within a comment.
      *
      * @param pattern The regular expression pattern to use.
-     * @param lineNo The line number in the file.
+     * @param line to check the relief pattern.
      * @return True if a match was found inside a comment.
      */
-    // suppress deprecation until https://github.com/checkstyle/checkstyle/issues/11166
-    @SuppressWarnings("deprecation")
-    private boolean matchesComment(Pattern pattern, int lineNo) {
-        final String line = getLine(lineNo - 1);
-
+    private static boolean matchesComment(Pattern pattern, String line) {
         final Matcher matcher = pattern.matcher(line);
-        return matcher.find()
-                && getFileContents().hasIntersectionWithComment(
-                        lineNo, matcher.start(), lineNo, matcher.end());
+        return matcher.find();
```

Comment:
this method become too small , please inline it to usage -  `pattern.matcher(line).find()`

Reply:
done